### PR TITLE
946 add installation of ca in system store for windows only

### PR DESCRIPTION
**Pull request by gregLibert**

### CODE REVIEW

File: pkg/certificate/store/store_windows.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+	"fmt"
+	"syscall"
+	"unsafe"
+)
+
+const (
+	CRYPT_E_NOT_FOUND = 0x80092004
+)
+
+// The functions bellow return an error even though they succeed.
```
Comment (peterjah) [inline pkg/certificate/store/store_windows.go:14]: These functions seems inspired from mkcert. maybe it worth to put the link here in comment

**Review by peterjah — APPROVED** [review]

File: pkg/certificate/store/store_windows.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+		certSyscall = (*syscall.CertContext)(unsafe.Pointer(certPtr))
+
+		// Parse cert
+		certBytes := (*[1 << 20]byte)(unsafe.Pointer(certSyscall.EncodedCert))[:certSyscall.Length]
+		certX509, err := x509.ParseCertificate(certBytes)
+
+		// Ignore parsing errors
+		if err != nil || certX509.SerialNumber == nil {
+			continue
```
Comment (Pierregvx) [inline pkg/certificate/store/store_windows.go:140]: could we add a warn logging here ?

**Review by Pierregvx — APPROVED** [review]

File: pkg/certificate/store/store_windows.go (hunk 2 above)
Comment (gregLibert) [reply pkg/certificate/store/store_windows.go:140]: What the point? What will we do with the warning ? The context here is that we are listing all the certificates of the ROOT trust store. if one of them is broken, we can do nothing...

File: pkg/certificate/store/store_windows.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+	"crypto/x509"
+	"fmt"
+	"syscall"
+	"unsafe"
+)
+
+const (
+	CRYPT_E_NOT_FOUND = 0x80092004
+)
+
```
Comment (Thykof) [inline pkg/certificate/store/store_windows.go:13]:
```suggestion

// inspired by https://github.com/FiloSottile/mkcert

```

File: pkg/certificate/store/store_windows.go (hunk 1 above)
Comment (Thykof) [reply pkg/certificate/store/store_windows.go:14]: like this https://github.com/massalabs/station/pull/954/files#r1252752318 ?

File: pkg/certificate/store/store_windows.go (hunk 2 above)
Comment (Pierregvx) [reply pkg/certificate/store/store_windows.go:140]: ok I see then I guess it is not useful indeed

**Review by thomas-senechal — APPROVED** [review]

File: pkg/certificate/store/store_windows.go (hunk 1 above)
Comment (Thykof) [reply pkg/certificate/store/store_windows.go:14]: we will do it in a following PR

**Pull request merged**